### PR TITLE
feat(curriculum): create design fundamentals quiz #56181

### LINKED ISSUE #56181: [HACKTOBERFEST 2024] - Design Fundamentals Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Design Fundamentals quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-design-fundamentals`.

**All quiz questions need to relate to one of these topics:**

- Common design terms
- BG/FG contrast
- visual heirarchy
- scale
- alignment
- whitespace
- working with images
- progressive enhancement
- user-centred designed
- user research, testing, and requirements
- best practises:
  - dark mode
  - bread crumbs
  - cards
  - infinite scroll
  - modals
  - progress indicators
  - shopping carts
- progressive disclosure
- deferred/lazy registration
- design briefs
- common design tools (e.g. figma, canva)

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by JoT8ng**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56181

### CODE REVIEW

File: sample.env
Comment (gikf) [inline sample.env:1]: This change is unrelated to the linked issue, please revert it.

Comment (JoT8ng) [reply sample.env:1]: I have updated it and reverted it

File: curriculum/challenges/english/25-front-end-development/quiz-design-fundamentals/66ed901ff45ce3ece4053eba.md [hunk 1]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+An iterative design process where designers design a product considering user requirements at every stage of the process
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is best practice contrast ratio for regular text for accessibility according to WCAG guidelines?
```
Comment (bbsmooth) [inline curriculum/challenges/english/25-front-end-development/quiz-design-fundamentals/66ed901ff45ce3ece4053eba.md:240]:
```suggestion
What is the minimum required contrast ratio for regular text for accessibility according to WCAG guidelines?
```

WCAG defines minimum requirements, not necessarily best practices.

**Review by naomi-lgbt — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-design-fundamentals/66ed901ff45ce3ece4053eba.md [hunk 2]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --text--
 
-Placeholder question
+What are some reasons why whitespace is important for good design?
```
Comment (Sembauke) [inline curriculum/challenges/english/25-front-end-development/quiz-design-fundamentals/66ed901ff45ce3ece4053eba.md:20]:
```suggestion
What are some reasons whitespace is important in good design?
```

**Review by Sembauke — APPROVED** [review]
